src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,462 @@
+    describe('for verified users', () => {
+        it('should allow to read the joiner', async() => {
+            // Given a part with a joiner, and a verified user
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            await createConnectedUser(CANDIDATE_EMAIL, CANDIDATE_NAME);
+
+            // When reading the joiner
+            const result: Promise<MGPOptional<Joiner>> = joinerDAO.read(partId);
+
+            // Then it succeeds
+            await expectAsync(result).toBeResolved();
+        });
+        it('should allow to create a joiner if there is a corresponding part', async() => {
+            // Given a verified user and an existing part
+            const createdPart: CreatedPart = await createPart({ createJoiner: false, signOut: true });
+            await reconnectUser(CREATOR_EMAIL);
+
+            // When creating the corresponding joiner, with the current user as creator
+            const result: Promise<void> = joinerDAO.set(createdPart.partId,
+                                                        { ...JoinerMocks.INITIAL, creator: createdPart.creator });
+
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        });
+        it('should forbid to create a joiner if there is no corresponding part', async() => {
+            // Given a verified user
+            const creator: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
+
+            // When creating a joiner with no corresponding part
+            const result: Promise<void> = joinerDAO.set('unexisting-part-id', { ...JoinerMocks.INITIAL, creator });
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid creating a joiner on behalf of another user', async() => {
+            // Given two users, including a malicious verified one, and an existing part
+            const createdPart: CreatedPart = await createPart({ createJoiner: false, signOut: true });
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the malicious user creates the corresponding joiner on behalf of the regular user
+            const result: Promise<void> = joinerDAO.set(createdPart.partId,
+                                                        { ...JoinerMocks.INITIAL, creator: createdPart.creator });
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid a to use a fake name in the list of candidates', async() => {
+            // Given a part and joiner, and a verified user
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the user adds themself to the list of candidates with a fake name
```

Comment:
users adds themselves*

Reply:
bah non y'a qu'un user qui s'ajoute !

Reply:
oui mais themself ça me fait pleurer x)

Reply:
ps: x4, au cas où j'oublie

Reply:
When the user is added? When the user joins the list of candidates?
On va pas remplacer un mot correct par un mot incorrect quand même :D

Reply:
when the user joins the list of candidates je dirais alors, si themselves pour un singulier te gêne et qu'on sais pas dire pluraliser le sujet comme on l'as fait dans d'autres phrases

Reply:
Plus de themself/ves dans ce fichier !